pkgs/applications/editors/spacevim/default.nix

```diff
@@ -0,0 +1,58 @@
+{ makeWrapper, vim_configurable, vimPlugins, fetchFromGitHub, writeTextDir
+, stdenv, spacevim_toml ? "" }:
+with stdenv;
+let
+  # At the moment, there is no way to write TOML given a attrset
+  # so the solution here is to allow a custom TOML file to be provided.
+  # The pkg includes a sensible TOML file for those that don't specify one.
+  config = if builtins.stringLength spacevim_toml == 0 then
+    "${builtins.readFile ./init.toml}"
+  else
+    spacevim_toml;
+  vim-customized = vim_configurable.customize {
+    name = "vim";
+    # Not clear at the moment how to import plugins such that
+    # SpaceVim finds them and does not auto download them to
+    # ~/.cache/vimfiles/repos
+    vimrcConfig.packages.myVimPackage = with vimPlugins; { start = [ ]; };
+  };
+  spacevimdir = writeTextDir "/init.toml" config;
+in mkDerivation {
+  name = "spacevim";
+  src = fetchFromGitHub {
+    owner = "SpaceVim";
+    repo = "SpaceVim";
+    rev = "c937c0e2fd37207c36c8c5e53b36c41d7222fee6";
+    sha256 = "141fl5g6i2h72dk5121v3mc0bwb812hd8qa5qw83jyz9q20jcsgn";
+  };
+
+  buildInputs = [ vim-customized makeWrapper ];
```

Comment:
At least `makeWrapper` belongs to [`nativeBuildInputs`](https://nixos.org/nixpkgs/manual/#ssec-stdenv-dependencies). However we should probably disable cross-compilation here.

Reply:
Can you help me understand the difference ?
Happy to make the change but also willing to learn.

Reply:
Moved to _nativeBuildInputs_

Reply:
Now that I think of it, `vim-customized` should probably be in `nativeBuildInputs` as well.

>  Happy to make the change but also willing to learn.

In `nativeBuildInputs` are all dependencies that are used *during* the derivation's build. When performing a cross-build, this is needed to make sure that only packages for the host's architecture are used for the build while all runtime dependencies (e.g. libraries in `buildInputs`) are built against the target's architecture.

A more detailed explanation is in the link I posted above (https://nixos.org/nixpkgs/manual/#ssec-stdenv-dependencies).